graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/method/MethodDataFetcherFactory.kt

```diff
@@ -29,12 +34,53 @@ import java.lang.reflect.Method
  * Resolving of method arguments is handled by the supplied [argument resolvers][ArgumentResolver].
  */
 class MethodDataFetcherFactory(
-    argumentResolvers: List<ArgumentResolver>,
+    private val argumentResolvers: List<ArgumentResolver>,
     private val parameterNameDiscoverer: ParameterNameDiscoverer = DefaultParameterNameDiscoverer()
 ) {
 
     private val resolvers = ArgumentResolverComposite(argumentResolvers)
 
+    fun checkInputArgumentsAreValid(method: Method, argumentNames: Set<String>) {
+        val inputArgumentResolvers = argumentResolvers.filterIsInstance<AbstractInputArgumentResolver>()
+        if (inputArgumentResolvers.isEmpty()) return
+
+        val bridgedMethod: Method = BridgeMethodResolver.findBridgedMethod(method)
+        val methodParameters: List<MethodParameter> = bridgedMethod.parameters.map { parameter ->
+            val methodParameter = SynthesizingMethodParameter.forParameter(parameter)
+            methodParameter.initParameterNameDiscovery(parameterNameDiscoverer)
+            methodParameter
+        }
+        val annotatedMethodParams = methodParameters.filter { it.hasParameterAnnotation(InputArgument::class.java) }
+        annotatedMethodParams.forEach { m ->
+            val selectedArgResolver = resolvers.getArgumentResolver(m)
+                ?: throw DataFetcherInputArgumentSchemaMismatchException("No suitable argument resolver found for method `${m.parameterName} on ${m.declaringClass}`.")
+            if (selectedArgResolver is AbstractInputArgumentResolver) {
+                val argName = selectedArgResolver.resolveArgumentName(m) ?: return@forEach
+                if (!argumentNames.contains(argName)) {
+                    val paramName = m.parameterName ?: return@forEach
+                    val arguments = if (argumentNames.isNotEmpty()) {
+                        "Found the following argument(s) in the schema: " + argumentNames.joinToString(prefix = "[", postfix = "]")
+                    } else {
+                        "No arguments on the field are defined in the schema."
+                    }
+
+                    when (selectedArgResolver) {
+                        is InputArgumentResolver -> throw DataFetcherInputArgumentSchemaMismatchException(
+                            "@InputArgument(name = \"$argName\") defined in ${method.declaringClass} in method `${method.name}` " +
+                                "on parameter named `$paramName` has no matching argument with name `$argName` in the GraphQL schema. " +
+                                arguments
+                        )
+                        is FallbackEnvironmentArgumentResolver -> throw DataFetcherInputArgumentSchemaMismatchException(
```

Comment:
Might be good to add a comment on when the fallback resolver is invoked in this scenario for future reference.

Reply:
will do, `FallbackEnvironmentArgumentResolver` covers the case where there is no `@InputArgument` as a last resort it uses the method parameter name, below named `name` :
```
       @DgsComponent
        class Fetcher {
            @DgsData(parentType = "Query", field = "hello")
            fun someFetcher(name: String?): String {
                return "Hello, ${name ?: "no name"}"
            }
        }
```

```
type Query {
   hello(name: String): String
}
```

It's covered by test case named: `when no @InputArgument fallback to method param name to match argument on schema`